packages/addons/tools/system-tools/package.mk

```diff
@@ -53,7 +53,10 @@ PKG_DEPENDS_TARGET="toolchain \
                     vim"
 
 if [ "${TARGET_ARCH}" = "x86_64" ]; then
-  PKG_DEPENDS_TARGET+=" efibootmgr st"
+  PKG_DEPENDS_TARGET+=" efibootmgr"
+  if [ "${DEVICE}" = "x11" -o "${DEVICE}" = "Generic-legacy" ]; then
+    PKG_DEPENDS_TARGET+=" st"
```

Comment:
does st just build with generic-x11 ? or does it build just not run at gbm?

Reply:
@CvH hmm, how (i.e. with which DEVICE) are generic addons built at jenkins, would st then even be built?
Might be easiest to split out X11-only stuff like st into a separate addon and make their build conditional on PROJECT/DEVICE (not sure if the latter is possible yet)

Reply:
Currently we build whatever is default at generic (gbm), chrome for example builds there too so currently there is just st as far I know. So I would rather keep it build for everything regardless it runs at gbm or not.

If we get more packaages that run just a this or that platform we might split them out into a addon with some magic that the users understand why it doesn't work (chrome for example).

Reply:
st is building on GBM by depending on the required X11 libs:

https://github.com/LibreELEC/LibreELEC.tv/blob/97ce049329e2444f534d4494edfe1432a83f85be/packages/addons/addon-depends/system-tools-depends/st/package.mk#L11

This seems to be working although st finally can only be started on X11.

I've created the commit with the assumption that (most?) kodi binary addons have to be build for the correct graphics stack and therefore we do need different repos at all for the unlikely case that there will be support for multiple Generic "devices" in the future.

But this can be dropped.

Reply:
at the moment everything builds well at "gbm" so I would assume we keep it that way, a dedicated x11 repo is something that needs to get maintained and build etc ... so I would prefer to keep it simple

Reply:
OK. Now st is build but only executed when X11 device is detected at runtime via /etc/os-release.

Reply:
hmm, now it's getting really complicated. I'd just split out st into a separate addon,

default.py of system-tools then only needs the "console only" dialog, default.py of st addon just contains the exec and we can make build of the st addon conditional on x11/legacy device like we do for various other addons - eg https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/mediacenter/kodi-binary-addons/screensaver.asterwave/package.mk#L21-L23

Reply:
Splitting out st should be done a different PR.